### ISSUE TITLE
Run-ALValidation fails with activated Telemetry

### ISSUE THREAD

**Issue by mmenninghaus** [microsoft/navcontainerhelper#2854]
**Describe the issue**
Run-ALValidation fails, if the Telemetry of BCContainerHelper is activated

**Scripts used to create container and cause the issue**
```
Run-ALValidation
```

**Full output of scripts**
```
BcContainerHelper version 4.0.10
Setting sandboxContainersAreMultitenantByDefault = False
BC.HelperFunctions emits usage statistics telemetry to Microsoft
...
2023-01-13T12:48:19.1336447Z Unexpected error while validating app. Error is: Exception setting "Message": "The type initializer for 'Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing.CoreEventSource' threw an exception."
2023-01-13T12:48:19.1345311Z at <ScriptBlock>, C:\Program Files\WindowsPowerShell\Modules\bccontainerhelper\4.0.10\TelemetryHelper.ps1: line 375
2023-01-13T12:48:19.1346387Z at TrackException, C:\Program Files\WindowsPowerShell\Modules\bccontainerhelper\4.0.10\TelemetryHelper.ps1: line 338
2023-01-13T12:48:19.1347449Z at New-BcContainer, C:\Program Files\WindowsPowerShell\Modules\bccontainerhelper\4.0.10\ContainerHandling\New-NavContainer.ps1: line 2416
2023-01-13T12:48:19.1348794Z at <ScriptBlock>, C:\Program Files\WindowsPowerShell\Modules\bccontainerhelper\4.0.10\AppHandling\Run-AlValidation.ps1: line 319
2023-01-13T12:48:19.1349832Z at <ScriptBlock>, C:\Program Files\WindowsPowerShell\Modules\bccontainerhelper\4.0.10\AppHandling\Run-AlValidation.ps1: line 465
2023-01-13T12:48:19.1350851Z at <ScriptBlock>, C:\Program Files\WindowsPowerShell\Modules\bccontainerhelper\4.0.10\AppHandling\Run-AlValidation.ps1: line 444
2023-01-13T12:48:19.1351872Z at <ScriptBlock>, C:\Program Files\WindowsPowerShell\Modules\bccontainerhelper\4.0.10\AppHandling\Run-AlValidation.ps1: line 428
2023-01-13T12:48:19.1352892Z at <ScriptBlock>, C:\Program Files\WindowsPowerShell\Modules\bccontainerhelper\4.0.10\AppHandling\Run-AlValidation.ps1: line 392
2023-01-13T12:48:19.1353927Z at Run-AlValidation, C:\Program Files\WindowsPowerShell\Modules\bccontainerhelper\4.0.10\AppHandling\Run-AlValidation.ps1: line 390
...
```

**Additional context**
- Disabling Telemetry solves the problem
- Full output contains sensitive data. We may provide it on a private channel

**Comment by freddydk** [comment]
Just heard this from another partner who found out that he was using an old version of the AZ powershell library
Could you try to see if that is the same?

**Comment by freddydk** [comment]
I have added some resilience to this area and if this fails, the command will just print out warnings instead of crashing.
It will also explain what the solution is. This is in prerelease now.

**Comment by HennieBink** [comment]
I have also this exception in the 4.0.11-preview869. 

ForEach-Object : Exception setting "Message": "The type initializer for 
'Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing.CoreEventSource' threw an exception."
At C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\4.0.11\TelemetryHelper.ps1:346 char:41
+                 "Microsoft","Partner" | ForEach-Object {
+                                         ~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [ForEach-Object], SetValueInvocationException
    + FullyQualifiedErrorId : ExceptionWhenSetting,Microsoft.PowerShell.Commands.ForEachObjectCommand

**Comment by freddydk** [comment]
Same think - caused by a conflict with another old PowerShell library probably - try to upgrade AZ

**Comment by freddydk** [comment]
I have added some resilience to this one as well in the next prerelease - thanks for notifying

**Comment by HennieBink** [comment]
With BcContainerHelper version 4.0.11-preview873 there is a warning instead of an error. But now we have another error later in the proces. See below.

Get-InstalledModule shows 5 modules. No AZ module??
Maybe the problem is that we don't have AZ installed?

Version              Name                                Repository           Description                              
-------              ----                                ----------           -----------                              
4.0.10               BcContainerHelper                   PSGallery            PowerShell module, which makes it easi...
1.0.0.8              DockerMsftProvider                  PSGallery            PowerShell module with commands for di...
1.4.8.1              PackageManagement                   PSGallery            PackageManagement (a.k.a. OneGet) is a...
2.2.5                PowerShellGet                       PSGallery            PowerShell module with commands for di...
3.29.2101.0          SharePointPnPPowerShellOnline       PSGallery            Microsoft 365 Patterns and Practices P...
Running AL pipeline
Done building onpremise versions.
Building for latest container...
BcContainerHelper version 4.0.11-preview873

New error: 
Docker Build failed with exit code 1
At C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\4.0.11\ContainerHandling\New-NavImage.ps1:660 char:25
+ ...              throw "Docker Build failed with exit code $LastExitCode"
+                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Docker Build failed with exit code 1:String) [], RuntimeException
    + FullyQualifiedErrorId : Docker Build failed with exit code 1

**Comment by freddydk** [comment]
could you share the full output of the second error - then I can see if that is related
Thanks

**Comment by freddydk** [comment]
What dotnet versions (not framework) do you have?
You can use:
```
dotnet --list-sdks
dotnet --list-runtimes
```
to list them

**Comment by HennieBink** [comment]
dotnet --list-sdks
dotnet --list-runtimes

Microsoft.NETCore.App 5.0.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.WindowsDesktop.App 5.0.6 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Problem is solved. I now use 4.0.11-preview875 (and update the license file). Thanks!

**Comment by freddydk** [comment]
The dependency is on dotnet 6.0, you can download that here: https://dotnet.microsoft.com/en-us/download/dotnet/6.0

**Comment by freddydk** [comment]
Fixed in 4.0.14
